providers/parseable/src/lib.rs

```diff
@@ -108,18 +108,25 @@ async fn run_query(query: &Query, config: &Config) -> Result<Vec<ProviderEvent>>
 
     let request = HttpRequest::post(url, query.to_string()).with_headers(headers);
     let response = make_http_request(request).await?;
-
     let body: Value = serde_json::from_slice(&response.body)?;
-    let Value::Array(arr) = body else {
-        return Err(Error::Other { message: String::from_utf8_lossy(&response.body).to_string() })
-    };
 
-    let mut rows = Vec::with_capacity(arr.len());
-    for value in arr {
-        rows.push(parse_row(value)?)
+    if response.status_code == 200 {
+        let Value::Array(arr) = body else { return Err(Error::Other {
+            message: format!(
+                "Expected an array, received: {}",
+                body.to_string()
+            )
+        })};
+        let mut rows = Vec::with_capacity(arr.len());
+        for value in arr {
+            rows.push(parse_row(value)?)
+        }
+        Ok(rows)
+    } else {
+        return Err(Error::Other {
+            message: String::from_utf8_lossy(&response.body).to_string(),
+        });
```

Comment:
Btw this is just returning error coming from Parseable.

For example the server response is like 
`Query Execution failed due to error in datafusion: SQL error: ParserError("Expected end of statement, found: r")`

Reply:
Makes sense! Maybe a comment might be in order for the next person to read this code?

Btw, you don't need the `return` statement here. The branch can just evaluate to the `Err`.

```suggestion
        // Return error message returned by Parseable API.
        Err(Error::Other {
            message: String::from_utf8_lossy(&response.body).to_string(),
        })
```